app/Http/Controllers/PetShelterController.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\PetShelterRequest;
+use App\Http\Resources\PetShelterResource;
+use App\Models\PetShelter;
+use Illuminate\Http\RedirectResponse;
+use Inertia\Inertia;
+use Inertia\Response;
+
+class PetShelterController extends Controller
+{
+    public function index(): Response
+    {
+        $shelters = PetShelter::all();
+
+        return Inertia::render("PetShelters/Index", [
+            "shelters" => PetShelterResource::collection($shelters)->resolve(),
+        ]);
+    }
+
+    public function store(PetShelterRequest $request): RedirectResponse
+    {
+        $data = $request->validated();
+
+        $addressData = collect($data)
+            ->only(["address", "city", "postal_code"])
+            ->toArray();
+
+        $shelter = PetShelter::query()->create($data);
+
+        if (!empty($addressData["address"])) {
+            $shelter->address()->create($addressData);
+        }
```

Comment:
Should be nice to move that into some action whuch has responsibility for creating PetShelter - and seperate creating `PetShelter` and `PetShelterAddress` - but for me that should be `json` field in `PetShelter`.

Reply:
https://rezakhademix.medium.com/what-are-laravel-action-classes-and-how-to-use-them-253217c43b06